subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyResolveTimeoutIntegrationTest.groovy

```diff
@@ -0,0 +1,204 @@
+        downloadedLibsDir = file('build/libs')
+        executer.withArgument("-D${SOCKET_TIMEOUT_SYSTEM_PROPERTY}=1000")
+    }
+
+    void blockingForProtocol(String protocol, HttpResource... resources) {
+        if (protocol == 'http') {
+            resources.each { it.expectGetBlocking() }
+        } else if (protocol == 'https') {
+            // https://issues.apache.org/jira/browse/HTTPCLIENT-1478
+            def keyStore = TestKeyStore.init(temporaryFolder.file('ssl-keystore'))
+            keyStore.enableSslWithServerCert(server)
+            keyStore.configureServerCert(executer)
+            server.expectSslHandshakeBlocking()
+        } else {
+            assert false
+        }
+    }
+
+    @Unroll
+    def "fails single build script dependency resolution if #protocol connection exceeds timeout"() {
+        given:
+        blockingForProtocol(protocol, moduleA.pom)
+        buildFile << """
+            buildscript {
+                ${mavenRepository(mavenHttpRepo)}
+
+                dependencies {
+                    classpath '${mavenModuleCoordinates(moduleA)}'
+                }
+            }
+        """
+
+        when:
+        fails('help')
+
+        then:
+        assertDependencyReadTimeout(moduleA)
+
+        where:
+        _ | protocol
+        _ | 'http'
+        _ | 'https'
+    }
+
+    @Unroll
+    def "fails single application dependency resolution if #protocol connection exceeds timeout"() {
+        given:
+        blockingForProtocol(protocol, moduleA.pom)
+        buildFile << """
+            ${mavenRepository(mavenHttpRepo)}
+            ${customConfigDependencyAssignment(moduleA)}
+            ${configSyncTask()}
+        """
+
+        when:
+        fails('resolve')
+
+        then:
+        assertDependencyReadTimeout(moduleA)
+        !downloadedLibsDir.isDirectory()
+
+        where:
+        _ | protocol
+        _ | 'http'
+        _ | 'https'
+    }
+
+    @Unroll
+    def "fails concurrent application dependency resolution if #protocol connection exceeds timeout"() {
+        given:
+        MavenHttpModule moduleB = publishMavenModule(mavenHttpRepo, 'b')
+        MavenHttpModule moduleC = publishMavenModule(mavenHttpRepo, 'c')
+        blockingForProtocol(protocol, moduleA.pom, moduleB.pom, moduleC.pom)
+
+        buildFile << """
+            ${mavenRepository(mavenHttpRepo)}
+            ${customConfigDependencyAssignment(moduleA, moduleB, moduleC)}
+            ${configSyncTask()}
+        """
+
+        when:
+        fails('resolve', '--max-workers=3')
+
+        then:
+        assertDependencyReadTimeout(moduleA)
+        assertDependencyReadTimeout(moduleB)
+        assertDependencyReadTimeout(moduleC)
+        !downloadedLibsDir.isDirectory()
+        where:
+        _ | protocol
+        _ | 'http'
+        _ | 'https'
+    }
+
+    def "tries dependency resolution from secondary repository if HTTP connection exceeds timeout"() {
```

Comment:
That's the use case that would need to fail in the other pull request.

Reply:
I've examined another PR, it's good there so I'm gonna remove this test.